core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -350,7 +354,8 @@ public TableMetadata updateSchema(Schema newSchema, int newLastColumnId) {
         currentSnapshotId, snapshots, snapshotLog, addPreviousFile(file, lastUpdatedMillis));
   }
 
-  public TableMetadata updatePartitionSpec(PartitionSpec newPartitionSpec) {
+  // newPartitionSpec's partition field IDs should have already been refreshed
```

Comment:
Spec field IDs don't need to be refreshed. `freshSpec` is used for two things:
1. If schema IDs are reassigned, then it rebuilds the spec to with the new schema
2. It sets the spec's ID to the one assigned in this method

Reply:
This comment is about the input parameter `PartitionSpec newPartitionSpec`, whose field ids has to be processed/refreshed so that `freshSpec` does not need to take care of that extra work.